### DISCUSSION TITLE
RequireJS and Search Insights

### DISCUSSION THREAD

**Discussion by giladmaker** [algolia/instantsearch#6367] [Q&A]
This project is importing the Algolia search-insights package. In regards to using RequireJS, search insight's docs state:
https://github.com/algolia/search-insights.js/blob/main/docs/requirejs.md

> When using [Require.js](https://requirejs.org/), the default UMD build might conflict and throw with a "Mismatched anonymous define() modules" message. This is a [known Require.js issue](https://requirejs.org/docs/errors.html#mismatch).
To work around this problem and ensure you capture all interactions occurring before the library is done loading, change ALGOLIA_INSIGHTS_SRC to point to the IIFE build, and load it via a <script> tag.
var ALGOLIA_INSIGHTS_SRC = "https://cdn.jsdelivr.net/npm/search-insights@2.17.2/dist/search-insights.iife.min.js";
...

The current project, however, does not seem to handle that scenario and always imports the non-iife module, resulting in the above error occurring when importing instantsearch using RequireJS.

Is there any plan to add RequireJS support that will cater for this issue? I could try raise a PR that falls back to importing the iife in the error handler of the current import, something like this (inside createInsightsMiddleware.ts):

```
        try {
          const script = document.createElement('script');
          script.async = true;
          script.src = ALGOLIA_INSIGHTS_SRC;
          script.onerror = () => {
            instantSearchInstance.emit('error', new Error(errorMessage));
          };
          document.body.appendChild(script);
          insightsClient.shouldAddScript = false;
        } catch (cause) {
            // ... if typeof require === 'function', try import the iife here, and only if that fails as well then proceed to below lines
          insightsClient.shouldAddScript = false;
          instantSearchInstance.emit('error', new Error(errorMessage));
        }
```

**Comment by Haroenv** [comment]
As a workaround, you can load the script yourself before loading InstantSearch, it will then use the `window.aa` instead of trying to load the UMD.
